app/services/github/profile_consumer.rb

```diff
@@ -1,4 +1,30 @@
 module Github
   class ProfileConsumer
+    attr_accessor :github_profile, :github_profile_response_code
+
+    GITHUB_RESPONSE_FIELDS = %w[login avatar_url html_url name company blog email bio twitter_username public_repos public_gists followers following updated_at location].freeze
+    PROFILE_FIELDS = %i[nickname avatar_url url name company blog email bio twitter_username public_repos public_gists followers following git_date location].freeze
+
+    def initialize(user_name)
+      github_response = ApiClient.new(user_name).call_for_github_profile_data
+      self.github_profile_response_code = github_response[:code]
+      self.github_profile = github_response[:body]
+    end
+
+    def call
+      return nil unless github_profile_response_code == 200
+
+      build_response
+    end
+
+    private
+
+    def build_response
+      assambled_profile = {}
+      (0...GITHUB_RESPONSE_FIELDS.size).each do |i|
+        assambled_profile[PROFILE_FIELDS[i]] = github_profile[GITHUB_RESPONSE_FIELDS[i]]
+      end
```

Comment:
A small refactor
```suggestion
      GITHUB_RESPONSE_FIELDS.each_with_index do |gh_field, i|
        assambled_profile[PROFILE_FIELDS[i]] = github_profile[gh_field]
      end
```

Reply:
Instead of having two separate arrays I think it would be cleaner to have a single constant like this
```ruby
FIELD_MAPPINGS = {
  nickname: "login",
  avatar_url: "avatar_url",
  url: "html_url",
  ...
}
```

and then this loop would look something like this
```suggestion
      FIELD_MAPPINGS.each do |profile_field, gh_field|
        assambled_profile[profile_field] = github_profile[gh_field]
      end
```

different from javascript, on ruby you can iterate on hashes
https://apidock.com/ruby/Hash/each

Reply:
And this one gets key and value of hash